pokepaste.py

```diff
@@ -194,6 +194,118 @@ def format_paste(pasteid, paste):
 
     return html_template['paste'].substitute(pasteid=encrypt_id(pasteid), mons=html_mons)
 
+def parse_pokemon(mon):
+    # Example input:
```

Comment:
In Python, there's a special format for information like this - a *docstring*. It looks like this:

```
def parse_pokemon(mon):
    """Parse a set in text form, and return a dictionary.

This is useful for doing foo and bar.

Used like so:

    >>> parse_pokemon('''ffffffffff (Gyarados) (M) @ Leftovers
    Ability: Intimidate
    Level: 57
    Shiny: Yes
    Happiness: 64
    EVs: 252 Atk / 4 SpD / 252 Spe
    Adamant Nature
    IVs: 21 HP / 7 Atk / 19 Def / 30 SpA / 0 SpD
    - Dragon Dance
    - Substitute
    - Waterfall
    - Crunch''')
    {'item': {'name': 'Leftovers', 'id': 234}, 'gender': 'M', 'species': 'Gyarados', 'nick': 'ffffffffff', 'ability': 'Intimidate', 'level': 57, 'shiny': True, 'happiness': 64, 'evs': {'hp': 0, 'atk': 252, 'def': 0, 'spa': 0, 'spd': 4, 'spe': 252}, 'ivs': {'hp': 21, 'atk': 7, 'def': 19, 'spa': 30, 'spd': 0, 'spe': 31}, 'moves': [{'name': 'Dragon Dance', 'moveid': 349, 'type': 'dragon', 'classification': 0}, {'name': 'Substitute', 'moveid': 164, 'type': 'normal', 'classification': 0}, {'name': 'Waterfall', 'moveid': 127, 'type': 'water', 'classification': 1}, {'name': 'Crunch', 'moveid': 242, 'type': 'dark', 'classification': 1}]}

Remember to do foo, bar and baz.
```